### PR TITLE
WIP: Fix concurrency bugs

### PR DESCRIPTION
**Pull request by alcarvalho**
Trying to solve concurrency bug #163, but the tests for these are getting stalled.

### CODE REVIEW

File: Moya/RxSwift/Moya+RxSwift.swift [hunk 1]
```diff
@@ -17,11 +17,16 @@ public class RxMoyaProvider<T where T: MoyaTarget>: MoyaProvider<T> {
...
+                objc_sync_enter(weakSelf)
+                let inFlight = weakSelf.inflightRequests[endpoint]
+                objc_sync_exit(weakSelf)
+                if let existingObservable = inFlight {
+                    return existingObservable
+                }
             }
 
-            let observable: Observable<MoyaResponse> =  AnonymousObservable { observer in
+            let observable: Observable<MoyaResponse> = create { observer in
```
Comment (ashfurrow) [inline Moya/RxSwift/Moya+RxSwift.swift]: I'm still new to RxSwift – could you describe the difference here?

**Comment by ashfurrow** [comment]
Build is stalling on simulator verification. We're seeing a similar problem on artsy/eidolon – I've restarted builds on master and this PR to try and isolate the problem to here or not.

File: Moya/RxSwift/Moya+RxSwift.swift (hunk 1 above)
Comment (alcarvalho) [inline Moya/RxSwift/Moya+RxSwift.swift]: There's none. It's just the usual, recommended, way of creating an Observable. It's also shorter. :wink:

**Comment by ashfurrow** [comment]
CircleCI is still stalling, but is succeeding on master. Not sure why. @alcarvalho: are the tests working on your local computer? Maybe the tests are hanging.

File: Moya/RxSwift/Moya+RxSwift.swift (hunk 1 above)
Comment (ashfurrow) [inline Moya/RxSwift/Moya+RxSwift.swift]: Cool.

**Comment by alcarvalho** [comment]
@ashfurrow yeah, the tests are stalling. I think we need to come up with a different way of testing concurrent requests for inflight logic. But I don't know how, yet.

**Comment by alcarvalho** [comment]
The tests are now passing on my machine. I am just not sure if this is a good solution.

**Comment by ashfurrow** [comment]
I'Il try to take a look at it this weekend – thanks again for all your hard work!

**Comment by alcarvalho** [comment]
Great! I hope this is working as intended.

**Comment by alcarvalho** [comment]
How can I help checking this, @ashfurrow?

**Comment by ashfurrow** [comment]
Ah, sorry this got away from me. I'll take a look ASAP. If you could, I'd love to have a changelog entry, and a comment on the PR explaining the changes would be awesome! :tada:

**Comment by ashfurrow** [comment]
Looks great! Thanks so much for your work, and your patience!

**Pull request merged**